capstone-app/src/app/modules/photos/components/image-container/image-container.component.ts

```diff
@@ -1,52 +1,74 @@
 import { Photo } from 'src/app/modules/photos/photo-template';
 import { PhotoFetcher } from 'src/app/modules/photos/services/photo-fetcher.service';
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input, OnChanges, Output, EventEmitter } from '@angular/core';
 
 @Component({
   selector: 'app-image-container',
   templateUrl: './image-container.component.html',
   styleUrls: ['./image-container.component.scss'],
 })
-export class ImageContainerComponent implements OnInit {
-  photos: Photo[] = [];
+export class ImageContainerComponent implements OnInit, OnChanges  {
+  displayPhotos: Photo[] = [];
+  originalPhotos: Photo[] = [];
   query: string;
-  limit: number;
   errorMessage: string;
 
-  constructor(private photosService: PhotoFetcher) {}
+  /**
+   * @Input limit - Receives count filter from parent on how many
+   * photos to display
+   * @Output limitChange - When an image cannot be displayed,
+   * tell the parent to update image count
+   */
+  @Input() limit: number;
+  @Output() limitChange = new EventEmitter<number>();
 
+  constructor(private photosService: PhotoFetcher) {}
   ngOnInit() {
+    this.limit = 10;
     this.getPhotos();
   }
 
+  ngOnChanges() {
+    this.limitPhotos(this.limit);
+  }
+
   getPhotos() {
     this.query = 'Boston';
-    this.limit = 8;
-    this.photosService.getPhotos(this.query, this.limit).subscribe((results) => {
-      for (const item of results.items) {
-        if (item) {
-          const photo = {
-            title: item.title,
-            link: item.link,
-            contextLink: item.image.contextLink,
-            height: item.image.height,
-            width: item.image.width,
-          };
-          this.photos.push(photo);
+    this.photosService.getPhotos(this.query, 10).subscribe(
```

Comment:
Maybe passing in this.limit instead of 10 in the .getPhotos(this.query, 10) method would be better because you already set limit to 10 in your ngOnInit().

Reply:
Good idea